### ISSUE TITLE
Brave constantly freezes on Linux

### ISSUE THREAD

**Issue by Nuc1eoN** [brave/brave-browser#6905]
I have a big issue which prevents me from using brave on linux.

Brave constantly freezes making it non-responsive for like 5 seconds. This does not happen on Windows 10 for me, however it does happen on my ArchLinux machine, as well as on my Android phone.

Is there any way I can debug this issue? The console does not seem to show any bug related output.

I'm on Brave 1.0 btw and tried previous versions too.

**Comment by ronanyeah** [comment]
I think Sync was causing this for me.
AMD Thinkpad, Arch Linux, Brave 78.1.0.1
https://community.brave.com/t/brave-freezes-up-frequently-and-regularly-video-attached/84731

**Comment by Nuc1eoN** [comment]
Yes, I think for me too, since I have like thousands of bookmarks. On my Android phone it hangs everytime I bookmark a webpage.

No idea how to fix it though

**Comment by zschuessler** [comment]
Possibly related: I went from Chromium/Chrome 78 -> 79 today and encountered this issue. This also affects Brave when I installed it to test for the issue:

https://support.google.com/chrome/thread/23283471?hl=en

See first comment in that thread for more debug info. Additionally I tried LinuxMint 17.2 and 17.3 OSs - both had Brave demonstrating the same issues Chromium & Chrome have with freezing constantly.

**Comment by mpaluta** [comment]
I'm on Mint and have not experienced this issue. It sounds like this is related to sync and not OS. Maybe you could rename it so it can be categorized appropriately and linked to other sync issues @Nuc1eoN

**Comment by Nuc1eoN** [comment]
@mpaluta No, on my Windows 10 systems it works perfectly fine.

**Comment by bsclifton** [comment]
Thanks for sharing the issue, @zschuessler 😄 Looks like it may indeed be a bug in upstream Chromium which is tracked here:
https://bugs.chromium.org/p/chromium/issues/detail?id=1036427&q=freeze

Folks here might try launching Brave with an extra command line argument:
`--no-xshm`

Can you give that a shot, @Nuc1eoN?

**Comment by Nuc1eoN** [comment]
@bsclifton I have tried the `--no-xshm` parameter but my brave still freezes intermittently on Linux

**Comment by bsclifton** [comment]
Thanks for giving that a go and reporting back, @Nuc1eoN 😄 

There's a similar (to this) issue opened which is specifically tracking this behavior with sync, for folks affected by that: https://github.com/brave/brave-browser/issues/6098

Definitely seems the issue you're seeing is not related to sync. If it's not working anyways, you may try turning off sync just to check

**Comment by Nuc1eoN** [comment]
@bsclifton Thanks, how would you suggest to disable sync? Is there a command? Or do I need to leave the chain?

**Comment by Nuc1eoN** [comment]
Ok it seems that my bug might actually correspond to #6098 !
It happens for about 15 seconds and indeed one of my CPU cores goes to 100% as long as it happens.

**Comment by Nuc1eoN** [comment]
Yeah reading through all of that this actually sounds exactly like my issue.

Still how can I exclude/confirm sync from testing? Can I disable it reliably somehow?

**Comment by bsclifton** [comment]
> @bsclifton Thanks, how would you suggest to disable sync? Is there a command? Or do I need to leave the chain? Or just uncheck to sync "Bookmarks"?
> 
> Not sure what would actually disable it reliably for testing.

Yup, leaving the chain would be the way to go. Let us know if that helps 😄

**Comment by Nuc1eoN** [comment]
Leaving the chain did not help. I have also tried to disable the OOP rasterization but it did not help either.

I think that #6098 and #7092 are both duplicates of my issue.

My GPU is an AMD RX470 btw with open source drivers.

**Comment by Nuc1eoN** [comment]
@bsclifton Ok insane, so after leaving the sync chain and actually _restarting_ my browser it seems to have gone away.

I will monitor this further but it's kinda insane that after months I might finally be able to use the brave browser on Linux. Of course it sucks that I cannot use my bookmarks...

And btw I am experiencing a similar issue on my Android smartphone, for instance everytime I add a bookmark my brave mobile freezes for like 10 to 15 seconds. And yes it also uses sync.

On Windows 10 this does not happen for me.

Again GPU is AMD RX470 and CPU is Ryzen 7 1700

**Comment by Gruntfuggly** [comment]
I am experiencing this since installing Brave on a new virtual machine. It was fine on an Ubuntu 16 VM with the same hardware but constantly locks up in my Ubuntu 18 VM. I'm not using Sync either.

**Comment by Nuc1eoN** [comment]
So I have used brave for about 24h now after leaving the chain and I'm yet to experience antoher freeze. Before that it happened every few minutes, so it is definitely "fixed".

@Gruntfuggly if you experience it without sync completely than I would argue it makes sense to keep two different bugreports, as my issue is clearly linked to sync.
I am currently waiting for a confirmation from #6098 if the issue is linked to sync.

**Comment by bsclifton** [comment]
@Gruntfuggly sounds like you're affected by https://bugs.chromium.org/p/chromium/issues/detail?id=1036427&q=freeze

Looks like bug's scope has been narrowed down and a work-around was provided. When starting Brave, launch via CLI and include `--no-xshm` as a param. If that works, likely that fix will be included in Chromium 80

**Comment by Gruntfuggly** [comment]
I tried using the command line and `--no-xshm` but it still happens.

**Comment by msundin** [comment]
I also have this issue where Brave freezes periodically. I'm 100% sure that it's an issue with sync (I have thousands of bookmarks) since every time I leave the sync chain on a system, the problem goes away completely. When I join the sync chain again, the problem reappears. On all systems.

I've confirmed this sync related problem in Brave on all these systems:
- Arch Linux
- Windows
- MacOS

I also have the problem on my Android phone but in another way it seems: Brave freezes every time i add a new bookmark if sync is enabled.

To sum up: **100% sync related**.

**Comment by rebron** [comment]
Closing. Complaint for this issue has been fixed latest chromium 79 update. Sync related is tracked with a different ticket.

**Comment by Nuc1eoN** [comment]
@rebron My issue is sync related.

**Comment by btlechowski** [comment]
No longer experience freezes. Fixed with Chromium 80.

Verification passed on

Brave | 1.3.113 Chromium: 80.0.3987.87 (Official Build) (64-bit)
-- | --
Revision | 449cb163497b70dbf98d389f54e38e85d4c59b43-refs/branch-heads/3987@{#801}
OS | Ubuntu 18.04 LTS

**Comment by ornitorrinho** [comment]
@bsclifton I experienced the same problem without being in a sync chain (I only use brave for desktop), and launching with the --no-xshm parameter seems to have fixed it. Is there any way to make incorporate this command to the application so I can launch it clicking on the Icon rather than launching it via terminal every time?

**Comment by btlechowski** [comment]
@dodekafonos please file another issue if you still have problems.

**Comment by Nuc1eoN** [comment]
@dodekafonos Edit your launch parameter to start with `--no-xshm`

**Comment by dodekafonos** [comment]
@Nuc1eoN Still trying to figure out **how** exactly do I do that lol

**Comment by Nuc1eoN** [comment]
@dodekafonos in most linux DEs just right click and go to properties. However it should be simple to google. You need to edit a .desktop file.

**Comment by dodekafonos** [comment]
I'm on Pop!_Os and there's no way to view properties with the right click, however I solved it installing a "menu editor" and adding the parameter through that program. Thanks!
